### PR TITLE
link team submission view to team workflow

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -**

This ticket addresses EDUCATOR-5182, which notes that cancelled Team ORAs do not show their cancellation info.
**What changed?**
The cauase was the xblock was linking the inappropriate workflow in that case.  I added  new method for the team_workflow_mixin to fetch the Teams workflow version of cancellation info and linked it into the submission_mixin where appropriate.

**Note** I realized while working on this that there don't seem to be any unit tests around the workflow_mixin currently, and created a new ticket for that work

**Developer Checklist**
- [ ] Reviewed the [release process](./release_process.md)
- [ ] Translations up to date
- [ ] JS minified, SASS compiled
- [ ] Test suites passing on Jenkins
- [ ] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [EDUCATOR-5182](https://openedx.atlassian.net/browse/EDUCATOR-5182)

FIY: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/xblock/team_workflow_mixin.py [hunk 1]
```diff
@@ -107,3 +108,30 @@ def get_team_workflow_status_counts(self):
...
         num_submissions = sum(item['count'] for item in status_counts)
         return status_counts, num_submissions
+
+    def get_team_workflow_cancellation_info(self, team_submission_uuid):
+        """
+        Returns cancellation information for a particular team submission.
+
+        :param team_submission_uuid: The team_submission identifier associated with the
+        sumbission to return information for.
+        :return: The cancellation information, or None if the submission has
```
Comment (jansenk) [inline openassessment/xblock/team_workflow_mixin.py:118]: nit: "the team submission"

File: openassessment/xblock/submission_mixin.py [hunk 2]
```diff
@@ -766,7 +766,12 @@ def submission_path_and_context(self):
             context['submit_enabled'] = submit_enabled
             path = "openassessmentblock/response/oa_response.html"
         elif workflow["status"] == "cancelled":
-            context["workflow_cancellation"] = self.get_workflow_cancellation_info(self.submission_uuid)
+            if self.teams_enabled:
+                context["workflow_cancellation"] = self.get_team_workflow_cancellation_info(
+                    workflow["team_submission_uuid"])
```
Comment (jansenk) [inline openassessment/xblock/submission_mixin.py:771]: close paren on new line

File: openassessment/xblock/test/test_submission.py [hunk 3]
```diff
@@ -1029,6 +1046,70 @@ def test_cancelled_submission(self, xblock):
...
+        )
+
+        self._assert_path_and_context(
+            xblock, 'openassessmentblock/response/oa_response_cancelled.html',
+            {
+                'text_response': 'required',
+                'file_upload_response': None,
+                'file_upload_type': None,
+                'allow_latex': False,
+                'submission_due': dt.datetime(2999, 5, 6).replace(tzinfo=pytz.utc),
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply openassessment/xblock/test/test_submission.py:1091]: This is copying off of the previous test cancellation test.  Not sure if there is a better way of pulling it off the xblock.

File: openassessment/xblock/test/test_submission.py [hunk 4]
```diff
@@ -1029,6 +1046,70 @@ def test_cancelled_submission(self, xblock):
...
+        xblock.teams_enabled = True
+        xblock.is_team_assignment = Mock(return_value=True)
+        anonymous_user_ids_for_team = ['Red Five', 'Bob', 'Alice', 'Chris']
+        xblock.get_anonymous_user_ids_for_team = Mock(
+            return_value=anonymous_user_ids_for_team
+        )
+        team_submission = xblock.create_team_submission(
+            ('a man must have a code', 'a man must also have a towel')
+        )
+        student_submission = dict(sub_api.get_submissions(
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply openassessment/xblock/test/test_submission.py:1064]: OrderedDict, unfortunately.  I tend to prefer having single-statement assignments for this rather than extra/unused variables when possible, but I am also used to working in ES6, which is a bit more readable for it, so I can split this up, I guess.

File: openassessment/xblock/team_workflow_mixin.py [hunk 5]
```diff
@@ -107,3 +108,30 @@ def get_team_workflow_status_counts(self):
...
+        not been cancelled.
+        """
+        cancellation_info = team_workflow_api.get_assessment_workflow_cancellation(
+            team_submission_uuid)
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply openassessment/xblock/team_workflow_mixin.py:122]: srry, trying to get used to python line continuations again :-p

File: openassessment/xblock/submission_mixin.py (hunk 2 above)
Comment (Andytr1) [reply openassessment/xblock/submission_mixin.py:771]: This is more of a stylistic nit:
consider using a ternary operator to get the proper workflow id so we can have fewer lines :)
```
uuid_to_use = workflow["team_submission_uuid"] if self.teams_enabled else self.submission_uuid
context["workflow_cancellation"] = self.get_team_workflow_cancellation_info(uuid_to_use)
```

File: openassessment/xblock/team_workflow_mixin.py [hunk 6]
```diff
@@ -107,3 +108,32 @@ def get_team_workflow_status_counts(self):
...
+            team_submission_uuid
+        )
+        if not cancellation_info:
+            return None
+
+        # Add the username of the staff member who cancelled the submission
+        cancellation_info['cancelled_by'] = self.get_username(cancellation_info['cancelled_by_id'])
+
+        # Add the date that the workflow was cancelled (in preference to the serialized date string)
+        del cancellation_info['created_at']
```
Comment (Andytr1) [inline openassessment/xblock/team_workflow_mixin.py:131]: Why are we clearing out the 'created_at' ?

Comment (muselesscreator) [reply openassessment/xblock/team_workflow_mixin.py:131]: copying from the workflow_mixin, so I'm not 100% sure.

File: openassessment/xblock/submission_mixin.py (hunk 2 above)
Comment (muselesscreator) [reply openassessment/xblock/submission_mixin.py:771]: different get_workflow methods as well, though.
